helix-stdx/src/faccess.rs

```diff
@@ -417,17 +439,61 @@ mod imp {
         Ok(())
     }
 
-    pub fn hardlink_count(p: &Path) -> std::io::Result<u64> {
+    fn file_info(p: &Path) -> io::Result<BY_HANDLE_FILE_INFORMATION> {
         let file = std::fs::File::open(p)?;
         let handle = file.as_raw_handle() as isize;
         let mut info: BY_HANDLE_FILE_INFORMATION = unsafe { std::mem::zeroed() };
 
         if unsafe { GetFileInformationByHandle(handle, &mut info) } == 0 {
-            Err(std::io::Error::last_os_error())
+            Err(io::Error::last_os_error())
         } else {
-            Ok(info.nNumberOfLinks as u64)
+            Ok(info)
         }
     }
+
+    pub fn hardlink_count(p: &Path) -> io::Result<u64> {
+        let n = file_info(p)?.nNumberOfLinks as u64;
+        Ok(n)
+    }
+
+    pub fn create_copy_mode(from: &Path, to: &Path) -> io::Result<std::fs::File> {
+        let sa = SECURITY_ATTRIBUTES {
+            nLength: std::mem::size_of::<SECURITY_ATTRIBUTES>() as u32,
+            lpSecurityDescriptor: SecurityDescriptor::for_path(from)?.sd,
+            bInheritHandle: FALSE,
+        };
+        let attributes = file_info(from)?.dwFileAttributes;
+
+        let from_file = std::fs::File::open(from)?;
+        let from_handle = from_file.as_raw_handle() as isize;
+
+        let to_path = std::fs::canonicalize(to)?;
+        let to_pathos = to_path.into_os_string();
+        let mut to_pathw: Vec<u16> = Vec::with_capacity(to_pathos.len() + 1);
+        to_pathw.extend(to_pathos.encode_wide());
+        to_pathw.push(0);
+
+        let handle = unsafe {
+            CreateFileW(
+                to_pathw.as_ptr(),
+                GENERIC_READ | GENERIC_WRITE,
+                FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE,
+                &sa,
+                CREATE_NEW,
+                attributes,
+                from_handle,
```

Comment:
I'm not entirely sure if I'm being redundant here. The documentation doesn't indicate which attributes are copied over from `hTemplateFile`.